src/Form/ThematicCommunityMembershipFilterType.php

```diff
@@ -0,0 +1,170 @@
+            ->add('joinedSince', DateType::class, ['required' => false, 'widget' => 'single_text', 'html5' => true])
+            ->add('joinedUntil', DateType::class, ['required' => false, 'widget' => 'single_text', 'html5' => true])
+            ->add('motivation', ChoiceType::class, [
+                'required' => false,
+                'choices' => ThematicCommunityMembership::MOTIVATIONS,
+                'choice_label' => function (string $choice) {
+                    return 'admin.thematic_community.membership.motivations.'.$choice;
+                },
+            ])
+            ->add('categories', EntityType::class, [
+                'required' => false,
+                'multiple' => true,
+                'class' => UserListDefinition::class,
+                'query_builder' => function (EntityRepository $er) {
+                    return $er->createQueryBuilder('uld')
+                        ->where('uld.type IN (:tc_types)')
+                        ->setParameter('tc_types', UserListDefinitionEnum::THEMATIC_COMMUNITY_CODES)
+                    ;
+                },
+            ])
+            ->add('expert', ChoiceType::class, [
+                'required' => false,
+                'expanded' => false,
+                'multiple' => false,
+                'placeholder' => 'Tous',
+                'choices' => [
+                    'Non' => 0,
+                    'Oui' => 1,
+                ],
+            ])
+            ->add('with_job', ChoiceType::class, [
+                'required' => false,
+                'expanded' => false,
+                'multiple' => false,
+                'placeholder' => 'Tous',
+                'choices' => [
+                    'Non' => 0,
+                    'Oui' => 1,
+                ],
+            ])
+            ->add('job', ChoiceType::class, [
+                'required' => false,
+                'expanded' => false,
+                'multiple' => true,
+                'choices' => JobEnum::JOBS,
+                'choice_label' => function (string $choice) {
+                    return $choice;
+                },
+            ])
+            ->add('with_association', ChoiceType::class, [
+                'required' => false,
+                'expanded' => false,
+                'multiple' => false,
+                'placeholder' => 'Tous',
+                'choices' => [
+                    'Non' => 0,
+                    'Oui' => 1,
+                ],
+            ])
+            ->add('sort', HiddenType::class, ['required' => false])
+            ->add('order', HiddenType::class, ['required' => false])
+        ;
+
+        if (\count($options['handled_communities']) > 1) {
+            $builder->add('thematicCommunities', EntityType::class, [
+                'required' => false,
+                'class' => ThematicCommunity::class,
+                'multiple' => true,
+                'query_builder' => function (EntityRepository $er) use ($options) {
```

Comment:
why do you not use the CT objects from `$options['handled_communities']` directly?
```php
'choices' => $options['handled_communities'],
```